fix different cases  between read() and write()

surfernsk opened the pull request:
Fix different cases between option names and key names in _comment_map

When read() and write() - options saves same cases

Preocts commented:
Please explain what this is solving for more clearly and show this working in a new unit test.

surfernsk commented:
By default, all keys in the sections are case-insensitive. for example, for option in parser ["section"] gives only the names of the optionxform'ed option keys. This means that the keys are in lowercase by default.

If you open an existing file in which the options are not in lowercase, for example:
````
; comment 1
; comment 1
[Section_1]

; CheckInputData comment
CheckInputData = true

; Option_2 comment
Option_2 = 34

; Option_3 comment
Option_3 = 36
````
Then after saving, we get the following result, some of the comments are gone, and the names of the options are converted to lowercase:
````
; comment 1
; comment 1
[Section_1]
; CheckInputData comment
checkinputdata = true
option_2 = 34
option_3 = 36
````

surfernsk committed:
fix different cases  between read() and write()

Fix different cases between option names and key names in _comment_map

When read() and write() - options saves same cases

surfernsk commented:
For clarification, I suggest the following:
1. By default, disable lowercase conversion.
2. If necessary, it will still be possible to override the "optionxform" method outside the "CommentedConfigParser" class.
````
from commentedconfigparser import CommentedConfigParser

config = CommentedConfigParser()

## if conversion to lowercase is required
config.optionxform = lambda option: option.lower()

config.read('test.ini')

##
# Some kind of manipulation of the parameters in the configuration.
##

with open('test_new.ini', 'w') as config_file:
    config.write(config_file)

surfernsk commented:
For clarification, I suggest the following:
1. By default, disable lowercase conversion.
2. If necessary, it will still be possible to override the "optionxform" method outside the "CommentedConfigParser" class.

Preocts reviewed (approved):
Great summary. Thank you for the additional details and the submission.
